automation/jenkins_build.sh

```diff
@@ -75,8 +75,13 @@ else
 	done
 fi
 
-# if the build is marked as failed, print failed images
+# if the build is marked as failed, print failed images and tidy up everything
 if [ $exitCode -eq 1 ]; then
 	echo Failed images: ${failedList[@]}
+	# we need to clean up all untagged images when builds fail.
+	# First remove all stopped containers.
+	docker rm -v $(docker ps --filter status=exited -q)
```

Comment:
Updated! it's hard to tell if a container created by builders or other people so I only stop containers related to the current target base images. But I don't expect anyone running anything on jenkins builders so cleaning up all stopped containers seems fine to me